docs/DeveloperGuide.md

```diff
@@ -1,37 +1,293 @@
 # Developer Guide
 
+- [Acknowledgements](#Acknowledgements)  
+- [Design & Implementation](#Design-&-Implementation)  
+  - [Architecture](#Architecture)
+  - [Implemented Features](#Implemented-Features)
+  - [Proposed Features](#Proposed-Features)
+- [Product scope](#Product-scope)
+- [User Stories](#User-Stories)
+- [Glossary](#Glossary)
+- [Instructions for manual testing](#Instructions-for-manual-testing)
+
+
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+The design of Developer Guide is referenced from the [AB-3 Developer Guide](http://se-education.org/addressbook-level3/DeveloperGuide.html).
+
+
+
+## Design & Implementation
+### Architecture
+#### Main components of the architecture
+![image info](./UmlDiagrams/Structure.png)  
+
+**Main** contains the main class called `CsProjPlanner`. It is responsible for:
+* At app launch: invokes methods to initialize the app, and gets prepared for receiving user input.
+* At shut down: invokes methods to store generated data.
+
+**UI** contains two classes `Constants` and `Response`. It is responsible for:
+* Store messages which will be displayed frequently.
+* Store magic numbers.
+* Handles the responses to give to user.  
+
+**Logic Component**  
+![image info](./UmlDiagrams/Logic.png)
+How the `Logic` component works:
+* When `CommandHandler` is called upon to execute a command, it creates `XYZCommandParser` object(e.g.,`AddProjectCommandParser`) which parses the user command and create a `XYZCommand` object (e.g., `AddProjectCommand`).
+* The created `XYZCommand` object is returned as a `Command` object.
+* All `XYZCommandParser` classes(e.g., `AddProjectCommandParser`, `AddTodoCommandParser`)inherit from the `Parser` interface.
+
+
+**Model Component**
+![image info](./UmlDiagrams/Model.png)
```

Comment:
Just for a better viewing experience, maybe you can use horizontal and vertical lines to connect the boxes of classs objects?